msteams/README.md

```diff
@@ -0,0 +1,162 @@
+`https://graph.microsoft.com/.default offline_access`. This requires the permissions for the
+client credential to be configured in Azure AD. It is important that the following
+permissions must be allowed for MS Graph API:
+
+- `Chat.Read`
+- `ChatMessage.Read`
+- `ChannelMessage.Read.All`
+- `Files.Read.All`
+- `offline_access`
+
+
+### App-only
+To use app-only authentication, set the following environment variables:
+
+- `MSTEAMS_GRAPH_AUTH_TYPE` to `application`
+- `MSTEAMS_GRAPH_TENANT_ID`
+- `MSTEAMS_GRAPH_CLIENT_ID`
+- `MSTEAMS_GRAPH_CLIENT_SECRET`
+- `MSTEAMS_USER_ID`
+
+These can be read from a .env file. See `.env-template`.
+
+The values for `MSTEAMS_TENANT_ID`, `MSTEAMS_CLIENT_ID` and `MSTEAMS_CLIENT_SECRET` come from
+Microsoft 365 admin. You must create an app registration in Microsoft 365 admin, and grant
+the appropriate permissions. The MSTEAMS_USER_ID represents the user ID of the individual who registered the app.
+To obtain the user ID, you can use the Microsoft Entra admin center Identity -> Users -> All Users menu and select
+your user. The user ID is the "Object ID" field. This information is essential for app-only authentication and is
+relevant to the limitations of the Microsoft Graph API search functionality.
+
+
+
+#### OAuth
+
+When using OAuth for authentication, the connector does not require any additional environment variables. Instead, the OAuth flow should occur outside of the Connector and Cohere's API will forward the user's access token to this connector through the `Authorization` header.
+
+With OAuth the connector will be able to search any Teams chat messages that the user has access to.
+
+To configure OAuth, follow the same steps in the Configuration section to create a Microsoft 365 App. You will also need to register a redirect URI on that app to `https://api.cohere.com/v1/connectors/oauth/token`.
+
+You can then register the connector with Cohere's API using the following configuration:
+Note: Your App key and App secret values correspond to `client_id` and `client_secret` respectively.
+
+```bash
+curl  -X POST \
+  'https://api.cohere.ai/v1/connectors' \
+  --header 'Accept: */*' \
+  --header 'Authorization: Bearer {COHERE-API-KEY}' \
+  --header 'Content-Type: application/json' \
+  --data-raw '{
+  "name": "MS Teams with OAuth",
+  "url": "{YOUR_CONNECTOR-URL}",
+  "oauth": {
+    "client_id": "{Your App CLIENT-ID}",
+    "client_secret": "{Your App CLIENT-SECRET}",
+    "authorize_url": "https://login.microsoftonline.com/{Your APP Tenant ID}/oauth2/v2.0/authorize"
+    "token_url": "https://login.microsoftonline.com/{Your APP Tenant ID}/oauth2/v2.0/token"
+    "scope": ".default offline_access"
+  }
+}'
+```
+
+
+To make a search requests to Microsoft Graph API, the connector needs to be authenticated with a token obtained using
+Oauth 2.0, and passed to the connector in the Authorization header with the Bearer schema.
```

Comment:
Can be open to interpretation, I would just mention that no more configuration is needed here and that after successfully registration, Cohere will take care of OAuth steps including passing in the correct headers